### PR TITLE
Update poa module schema validation and endpoint

### LINKED ISSUE #8944: Add additional schema validation to poa module
### Expected behavior

Schema validation should be used when available

### Actual behavior

 In two locations in the proof of authority module, checks are done manually when schema 
 validation could be used instead. 

`endpoint.ts`:  the parameter given to  getValidator  is checked in the following way: 
```
 const  {  address  }  =  context.params; 
 if  (  typeof  address  !==  'string'  )  { 
 throw  new  Error  (  'Parameter address must be a string.'  ); 
 } 
 cryptoAddress.validateLisk32Address(address); 
```

Check in  getValidator 
 (  lisk-sdk/framework/src/modules/poa/endpoint.ts#33-37  ) 
```
 validator.validate(getValidatorRequestSchema,  address) 
```
 
 In  commands/update_authority.ts  the length of the command’s  newValidators  list is 
 checked in the following way: 
```
 if  (newValidators.length  <  1  ||  newValidators.length  >  MAX_NUM_VALIDATORS)  { 
 throw  new  Error  ( 
 `newValidators length must be between 1 and  ${  MAX_NUM_VALIDATORS  } 
 (inclusive).`  , 
 ); 
 } 
```

Length check in  updateAuthority 
 (  lisk-sdk/framework/src/modules/poa/commands/update_authority.ts#33-37  ) 

### Steps to reproduce

Check poa logic

### Which version(s) does this affect? (Environment, OS, etc...)
v6.0.0

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #8944 and #8945, #8962 and #8965

### How was it solved?

- Updated endpoint
- Schema validation for update authority command is done in [PR](https://github.com/LiskHQ/lisk-sdk/pull/8953/files)
- Use constants

### How was it tested?

Check tests

### CODE REVIEW

File: framework/src/modules/poa/endpoint.ts [hunk 1]
```diff
@@ -30,10 +32,10 @@ export class PoAEndpoint extends BaseEndpoint {
 
 	public async getValidator(context: ModuleEndpointContext): Promise<Validator> {
 		const validatorSubStore = this.stores.get(ValidatorStore);
-		const { address } = context.params;
-		if (typeof address !== 'string') {
-			throw new Error('Parameter address must be a string.');
-		}
+		const address = context.params.address as string;
+
+		validator.validate(getValidatorRequestSchema, context.params);
```
Comment (ishantiw) [inline framework/src/modules/poa/endpoint.ts:37]: I think validate should be called before we access address param
```suggestion
                validator.validate(getValidatorRequestSchema, context.params);
                
		const address = context.params.address as string;
```

**Review by Phanco — APPROVED** [review]

File: framework/src/modules/poa/commands/register_authority.ts [hunk 2]
```diff
@@ -50,20 +49,12 @@ export class RegisterAuthorityCommand extends BaseCommand {
...
-				error: err as Error,
-			};
-		}
 
 		if (!POA_VALIDATOR_NAME_REGEX.test(name)) {
 			throw new Error(`Name does not comply with format ${POA_VALIDATOR_NAME_REGEX.toString()}.`);
 		}
 
-		const nameExists = await this.stores.get(NameStore).has(context, Buffer.from(name));
+		const nameExists = await this.stores.get(NameStore).has(context, Buffer.from(name, 'utf-8'));
```
Comment (ishantiw) [inline framework/src/modules/poa/commands/register_authority.ts:57]: I think by default its utf-8 so we don't have to mention it

File: framework/src/modules/poa/constants.ts [hunk 3]
```diff
@@ -43,3 +42,7 @@ export const defaultConfig = {
 export const KEY_SNAPSHOT_0 = utils.intToBuffer(0, 4);
 export const KEY_SNAPSHOT_1 = utils.intToBuffer(1, 4);
 export const KEY_SNAPSHOT_2 = utils.intToBuffer(2, 4);
+export const SUBSTORE_PREFIX_VALIDATOR = 0;
+export const SUBSTORE_PREFIX_CHAIN = 1;
+export const SUBSTORE_PREFIX_NAME = 2;
+export const SUBSTORE_PREFIX_SNAPSHOT = 3;
```
Comment (ishantiw) [inline framework/src/modules/poa/constants.ts:48]: can we rename them to `SUBSTORE_PREFIX_VALIDATOR_INDEX` as these values varies from the values given in the LIP but these indexes will be used to compute the given values in the LIP

File: framework/test/unit/modules/poa/commands/register_authority.spec.ts [hunk 4]
```diff
@@ -113,6 +115,69 @@ describe('RegisterAuthority', () => {
...
+
+		it(`should throw error when generator key longer than ${LENGTH_GENERATOR_KEY}`, () => {
+			expect(() =>
+				validator.validate(registerAuthorityCommand.schema, {
+					...registerAuthorityTransactionParams,
+					generatorKey: utils.getRandomBytes(LENGTH_GENERATOR_KEY + 1),
+				}),
+			).toThrow(`Property '.generatorKey' maxLength exceeded`);
+		});
+
```
Comment (ishantiw) [inline framework/test/unit/modules/poa/commands/register_authority.spec.ts:180]: can we move this outside `describe('verify')` and instead create a new one `describe('verify schema')` similar to other commands

File: framework/test/unit/modules/poa/commands/update_generator_key.spec.ts [hunk 5]
```diff
@@ -87,6 +102,22 @@ describe('UpdateGeneratorKey', () => {
...
+		});
+
+		it(`should throw error when generator key longer than ${LENGTH_GENERATOR_KEY}`, () => {
+			expect(() =>
+				validator.validate(updateGeneratorKeyCommand.schema, {
+					generatorKey: utils.getRandomBytes(LENGTH_GENERATOR_KEY + 1),
+				}),
+			).toThrow(`Property '.generatorKey' maxLength exceeded`);
+		});
+
```
Comment (ishantiw) [inline framework/test/unit/modules/poa/commands/update_generator_key.spec.ts:120]: same here, if we can move it to `describe('verify schema')` block

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: framework/src/modules/poa/commands/register_authority.ts (hunk 2 above)
Comment (mitsuaki-u) [reply framework/src/modules/poa/commands/register_authority.ts:57]: The audit report pointed out that in some places we use `Buffer.from` and some places `Buffer.from(..,'utf8') and that should be consistent which is why I updated it here.

File: framework/test/unit/modules/poa/commands/update_generator_key.spec.ts (hunk 5 above)
Comment (ishantiw) [reply framework/test/unit/modules/poa/commands/update_generator_key.spec.ts:120]: I think this comment is not resolved

**Review by ishantiw — COMMENTED** [review]
LGTM, one comment that was missed I think

**Review by ishantiw — APPROVED** [review]

**Pull request merged**